script.js

```diff
@@ -0,0 +1,35 @@
+const userInput = document.getElementById('user-input');
+const checkBtn = document.getElementById('check-btn');
+const clearBtn = document.getElementById('clear-btn');
+const resultsDiv = document.getElementById('results-div');
+
+function checkValidNumber(input) {
+  const countryCode = '^(1\\s?)?';
+  const areaCode = '(\\([0-9]{3}\\)|[0-9]{3})';
+  const spacesDashes = '[\\s\\-]?';
+  const phoneNumber = '[0-9]{3}[\\s\\-]?[0-9]{4}$';
+  const phoneRegex = new RegExp(`${countryCode}${areaCode}${spacesDashes}${phoneNumber}`);
+  resultsDiv.innerHTML = `${phoneRegex.test(input) ? 'Valid' : 'Invalid'} US number: ${input}`;
+}
+
+checkBtn.addEventListener('click', () => {
+  const input = userInput.value;
+
+  if (input.length === 0) {
+    alert('Please provide a phone number');
+  } else {
+    checkValidNumber(input);
+    userInput.value = '';
+  }
+});
+
+userInput.addEventListener('keydown', (e) => {
+  if (e.key === 'Enter') {
+    checkValidNumber(userInput.value);
+    userInput.value = '';
+  }
+});
+
+clearBtn.addEventListener('click', () => {
+  resultsDiv.innerHTML = '';
+});
```

Comment:
- [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  // As an Example I added only one of your Event Listeners add the rest of them like this
  
  const clearBtn = document.getElementById('clear-btn');
  const resultsDiv = document.getElementById('results-div');
  
  clearBtn.addEventListener('click', () => {
    resultsDiv.innerHTML = '';
  });
})
```